src/rt/io.rs

```diff
@@ -294,6 +295,20 @@ impl<T: ?Sized + Read + Unpin> Read for &mut T {
     deref_async_read!();
 }
 
+impl<P> Read for Pin<P>
+where
+    P: DerefMut,
```

Comment:
I notice in Tokio, it's `P: DerefMut + Unpin`. Do you know why this should be different, or if it should be the same as Tokio?

Reply:
My guess is that they wanted to avoid having to use unsafe due to `Pin::as_deref_mut` being unstable (or not existing at all, it seems like that impl dates verbatim back to 2019), or maybe the semantics of Pin weren't as fleshed out at the time(?)

Reply:
In any case, `Future` has [this same bound](https://doc.rust-lang.org/nightly/src/core/future/future.rs.html#118-127) for its impl for Pin.